logger.go

```diff
@@ -239,74 +394,53 @@ func (l *Log) Flush() {
 	wg.Wait()
 }
 
+func (log *Log) detachedDone() {
+	deps := func() []*Log {
+		log.request.span.dependentLock.Lock()
+		defer log.request.span.dependentLock.Unlock()
+		deps := make([]*Log, 0, len(log.shared.ActiveDetached))
+		for _, dep := range log.shared.ActiveDetached {
+			deps = append(deps, dep)
+		}
+		return deps
+	}()
+	for _, dep := range deps {
+		_ = dep.recursiveDone(false)
+	}
+}
+
 // Marks this request as boring.  Any log at the Alert or
 // Error level will mark this request as not boring.
 func (l *Log) Boring() {
-	requestBoring := atomic.LoadInt32(&l.request.boring)
+	requestBoring := atomic.LoadInt32(&l.request.span.boring)
 	if requestBoring != 0 {
 		return
 	}
-	l.request.base.Boring(true)
+	l.request.span.base.Boring(true)
 	// There is chance that in the time we were sending that
 	// boring=true, the the request became un-boring. If that
 	// happened, we can't tell if we're currently marked as
 	// boring, so let's make sure we're not boring by sending
 	// a false
-	requestStillBoring := atomic.LoadInt32(&l.request.boring)
+	requestStillBoring := atomic.LoadInt32(&l.request.span.boring)
 	if requestStillBoring != 0 {
-		l.request.base.Boring(false)
+		l.request.span.base.Boring(false)
 	}
-	l.enableFlushTimer()
+	l.hasActivity(true)
```

Comment:
Maybe the condition in line `426` should be changed to `if requestStillBoring == 0 {`. This is because `0` value indicates that the request has become non-boring (at least that is what I understand).